rasgotransforms/rasgotransforms/transforms/metric/metric.yaml

```diff
@@ -0,0 +1,56 @@
+name: metric
+type: insight
+tags:
+  - insight
+description: Calculate metric values given a metric definition
+arguments:
+  timestamp:
+    type: column
+    description: The date/datetime column by which the metric will be aggregated
+  target_sql:
+    type: column
+    description: The target column used to calculate the metric, can also be a sql string '(e.g. cast(target_column as string)')
+  aggregation_type:
+    type: string
+    description: the Aggregation method used to calculate the metric (e.g. 'sum', 'count_distinct', 'max')
+  time_grain:
+    type: string
```

Comment:
i think we support `    type: date_part` for this

Reply:
The only issue is that this template will only support 'day', 'week', 'month', 'quarter', and 'year' while `date_part` has a bunch of other date part options.